client/components/editor/teaching-elements/Assessment/ImageMultipleChoice.vue

```diff
@@ -106,45 +115,36 @@ export default {
       return new Promise((resolve, reject) => {
         this.resetErrorMessages(dataset.index);
         if (imageFile.size > 100000) {
-          // eslint-disable-next-line prefer-promise-reject-errors
-          return reject('The image is too large. Maximum allowed image size: 100 kB');
+          return reject(new Error('The image is too large. Maximum allowed image size: 100 kB'));
         }
         return resolve(imageFile);
       })
         .then(() => blobToBase64String(imageFile))
         .then(base64String => {
-          answers[dataset.index] = base64String;
-          return this.getImageDimensions(base64String);
+          let imageMimeType = imageFile.type;
+          answers[dataset.index] = getImageDataUrl(base64String, imageMimeType);
+          return this.getImageDimensions(base64String, imageMimeType);
         })
         .then(({ width, height }) => {
           if (width > maxImageDimension || height > maxImageDimension) {
-            // eslint-disable-next-line prefer-promise-reject-errors
-            return Promise.reject(
+            return Promise.reject(new Error(
               `Incorrect image width and/or height. Maximum allowed image dimensions:
                ${maxImageDimension}x${maxImageDimension}`
-            );
+            ));
           }
         })
-        .catch(errorMsg => {
-          this.addErrorMessage(dataset.index, errorMsg);
+        .catch(error => {
           answers[dataset.index] = '';
+          this.addErrorMessage(dataset.index, error.message);
         })
-        .then(() => {
-          this.update({ answers });
-        });
-    },
-    getImageDimensions(base64String) {
-      return new Promise((resolve, reject) => {
-        let i = document.createElement('img');
-        i.onload = () => {
-          resolve({ width: i.width, height: i.height });
-        };
-        i.src = this.getSrc(base64String);
-      });
+        .then(() => { this.update({ answers }); });
     },
-    getSrc(base64String) {
-      if (base64String === '') return './assets/img/no-image.png';
-      return `data:image/jpg;base64,${base64String}`;
+    getImageDimensions(base64String, mimeType) {
+      return new Promise((resolve, reject) => extractImageDimensions(
+        base64String,
+        mimeType,
+        resolve
+      ));
```

Comment:
Why is this still a method? 😕 
It is merely promisification of `extractImageDimensions` and does not use `this` at all...